### PR TITLE
Adding a Resource document to outline the current release

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: doc/c_sdk_resource_information.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# C-SDK Resource Information
+
+Detailed here are the captured the C SDK Resource constraints for each protocol.
+
+All test are executed on an ubuntu 18.04.1 image with GCC version 7.3.0 using C-SDK version 1.2.12.
+
+## On Disk Binary Information
+
+This is the size of the binary built with the c sdk for each protocol.  The appication setups the SDK and sends 1 telemetry message.
```
Comment (jebrando) [inline doc/c_sdk_resource_information.md]: >appication [](start = 76, length = 10)

sp application

Comment (jebrando) [inline doc/c_sdk_resource_information.md]: >setups [](start = 87, length = 6)

sets up

File: doc/c_sdk_resource_information.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# C-SDK Resource Information
+
+Detailed here are the captured the C SDK Resource constraints for each protocol.
```
Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >are the captured [](start = 14, length = 16)

capture

Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >constraints [](start = 50, length = 11)

maybe utilization? or consumption?

File: doc/c_sdk_resource_information.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+All test are executed on an ubuntu 18.04.1 image with GCC version 7.3.0 using C-SDK version 1.2.12.
```
Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >test [](start = 4, length = 4)

tests

File: doc/c_sdk_resource_information.md (hunk 1 above)
Comment (tameraw) [inline doc/c_sdk_resource_information.md]: > c sdk [](start = 45, length = 6)

This is the IoT Hub client only, no DPS Client. Correct? We should clarify that and add section for DPS (and maybe keep as TBD or WIP till we get the data).

File: doc/c_sdk_resource_information.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+| MQTT                    |      1        |      15,058      | 172
+| MQTT With Web Sockets   |      1        |      15,753      | 234
+| AMQP                    |      1        |      27,335      | 1,676
+| AMQP With Web Sockets   |      1        |      28,809      | 1,882
+
+## Network traffic Information
+
+Network packets measure the number of bytes that are sent form the C-SDK with the specified payload size.  This value excludes the connection and disconnection headers.
+
+| Protocol                | Messages Payload | Bytes Sent | Number Sends | Received Bytes | Number Received
```
Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >Number Sends [](start = 60, length = 12)

What does this number refer to?

Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >Received Bytes | Number Received [](start = 75, length = 32)

Not clear on those numbers as well.
Can you add some underscript and define what those columns mean below?

File: doc/c_sdk_resource_information.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+Detailed here are the captured the C SDK Resource constraints for each protocol.
+
+All test are executed on an ubuntu 18.04.1 image with GCC version 7.3.0 using C-SDK version 1.2.12.
+
+## On Disk Binary Information
+
+This is the size of the binary built with the c sdk for each protocol.  The appication setups the SDK and sends 1 telemetry message.
+
+| Protocol                | Size In Bytes
```
Comment (tameraw) [inline doc/c_sdk_resource_information.md]: >Size In Bytes [](start = 28, length = 13)

Would be nice to break this down for the following configs:
- SDK with everything in.
- With everything minus uploadToBlob
- With everything minus uploadToBlob & Logging

And include the cmake flag that builds each above in the footnotes.

File: doc/c_sdk_resource_information.md (hunk 1 above)
Comment (jebrando) [reply doc/c_sdk_resource_information.md]: Yes, DPS can be added later.

---
In reply to: [246860132](https://github.com/Azure/azure-iot-sdk-c/pull/802#discussion_r246860132) [](ancestors = 246860132)

**Pull request merged**